### ISSUE TITLE
Can the generated .gitignore files get a clear copyright header?

### ISSUE THREAD

**Issue by CharString** [toptal/gitignore.io#542]
It is unclear now under what license the .gitignore files can be copied and used.
#155 and #229 already brought it up. But the vague pointing to a LICENSE.md that doesn't even contain the copyright holder doesn't really clear it up. And the mention of "business requirements" in issues like #502 doesn't make the issue any better.

Just a header like https://reuse.software/tutorial/#insignificant-files that shows that the entity that produced the template puts the template in the public domain (or clearly reusable under a different permissive license) would be enough.

**Comment by Dragon1573** [comment]
This repository ([toptal/gitignore.io](https://github.com/toptal/gitignore.io)) and the `.gitignore` Templates Repository ([toptal/gitignore](https://github.com/toptal/gitignore)) are all licensed under *the MIT License*. Maybe attach it on our own is a walkaround?

**Comment by CharString** [comment]
But the source material is CC0: https://github.com/github/gitignore/blob/master/LICENSE

So if all that gitignore.io does is spit out those snippets, then prepending the `# Created by https://www.toptal.c...` bit with

```
# SPDX-FileCopyrightText: 2021 Toptal, LLC
#
# SPDX-License-Identifier: CC0-1.0
```
would be the clearest. It will declare that any additions that Toptal might have made to the CC0 licensed material by GitHub, Inc are also deemed CC0.

Licensing the templates under MIT could raise questions on whether that license requires any software project that uses the .gitignore template to reproduce the Copyright notice in their output.

To be clear I am just talking about the templates and the output the gitignore.io tool generates. All code of the tool itself can of course remain MIT.

**Comment by deniskaber** [comment]
Hello @CharString. Thank you for this suggestion!
I am going to bring this request to a team's attention. We will investigate and set priority for it.

**Comment by Dragon1573** [comment]
> But the source material is [CC0](https://github.com/github/gitignore/blob/master/LICENSE).

@toptal is currently using [toptal/gitignore](https://github.com/toptal/gitignore) to power the website instead of [github/gitignore](https://github.com/github/gitignore). It was migrated since [2017-11-07](https://blog.joeblau.com/gitignore-io-template-fork). 

Because [toptal/gitignore](https://github.com/toptal/gitignore) has no "fork relationships" with [github/gitignore](https://github.com/github/gitignore) (I can't find @toptal's repository in **Insights -> Forks** tab), I can't sure if @toptal's repository is a "duplicate without fork" of @github's. 

@github's repository DOES use _CC0 License_, but @toptal's uses _MIT License_ instead. It's still arguable for the license ...

Could @deniskaber explain this, please?

**Comment by CharString** [comment]
@Dragon1573 Well, not entirely... It looks like a mixture of CC0 and merged contributions

https://github.com/toptal/gitignore/blob/master/.github/scripts/sync-github.sh

**Comment by Dragon1573** [comment]
@CharString, With your instructions, I find a **GitHub Workflow** [configuration file](https://github.com/toptal/gitignore/blob/master/.github/workflows/lint.yml) in `toptal/gitignore`. 

I've searched several informations about the CC0 License. Some of them said that the CC0 License is very similar to the Unlicense. I have heard about the infectivity of GPL License Series, but I haven't heard about that of CC0. So, MAYBE using the MIT License instead of using CC0 License is acceptable?

**Comment by CharString** [comment]
CC0 isn't copyleft like GPL, so adaptations aren't required to be released under CC0... But... the source material under CC0 does constitute "prior art", at least in view of most patent systems. I'm not a lawyer, I don't know what it means for copyright.

My layman view, is that CC0 boils down to that we can regard is as, say, Shakespear's Hamlet. We can freely use it as it lies in the public domain. Taking a copy of Hamlet and slapping your own copyright notice in front, that only makes sense if you edit Hamlet and make significant adaptations.

My point with MIT is that it has this lline

> The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

I doubt that we will want to require any software that uses a .gitignore for the gitignore.io service to be able to produce said notice. And it's even unclear whether that's actually required. CC0 doesn't have that requirement.

Bottom line is in that "unclear". As it is, just from the output from gitignore.io it is not clear under what terms people can use the output. And come to think of it, it isn't clear under what license the contributions from others that @joeblau has pulled in on top of the CC0 material from Github Inc can be used.  Am I required to git blame and contact all copyright holders? Using gitignore.io is supposed to safe me time, not add another legal chore.

The [reuse spec](https://reuse.software/) tries to make it easy to be clear about how any file from your project may be reused. It's a great read to get familiar with the problem.

**Comment by mcarneiro** [comment]
Hello, we just merged an explicit license information about generated files in the project readme file (being under [CC0](https://creativecommons.org/publicdomain/zero/1.0/)). I believe that fixes this issue.

**Comment by CharString** [comment]
Not a  SPDX header.. but definitely good enough for me. I can add the header myself. At least it is clear now I can do so!
Thanks very much.
